src/components/Transactions/TransactionContract.tsx

```diff
@@ -0,0 +1,194 @@
+                    : { color: 'inherit', prefix: '-' })}
+                  value={fromBaseUnit(txDetails.value, txDetails.precision)}
+                  symbol={txDetails.symbol}
+                  maximumFractionDigits={6}
+                />
+              )}
+            </Flex>
+          </Flex>
+        </Flex>
+      </Flex>
+      <Collapse in={isOpen} unmountOnExit>
+        <SimpleGrid gridTemplateColumns='repeat(auto-fit, minmax(180px, 1fr))' spacing='4' py={6}>
+          <Row variant='vertical'>
+            <Row.Label>
+              <Text translation='transactionRow.date' />
+            </Row.Label>
+            <Row.Value>{dayjs(Number(txDetails.tx.blockTime) * 1000).format('LLL')}</Row.Value>
+          </Row>
+          <Row variant='vertical'>
+            <Row.Label>
+              <Text translation='transactionRow.txid' />
+            </Row.Label>
+            <Row.Value>
+              <Link
+                isExternal
+                color='blue.500'
+                href={`${txDetails.explorerTxLink}${txDetails.tx.txid}`}
+              >
+                <MiddleEllipsis address={txDetails.tx.txid} />
+              </Link>
+            </Row.Value>
+          </Row>
+
+          <Row variant='vertical' hidden={!txDetails.tradeTx}>
+            <Row.Label>
+              <Text translation={'transactionRow.amount'} />
+            </Row.Label>
+            <Row.Value>
+              <Amount.Crypto
+                value={fromBaseUnit(
+                  txDetails.sellTx?.value ?? '0',
+                  txDetails.sellAsset?.precision ?? 18
+                )}
+                symbol={txDetails.sellAsset?.symbol ?? ''}
+                maximumFractionDigits={6}
+              />
+              <Text translation='transactionRow.for' />
+              <Amount.Crypto
+                value={fromBaseUnit(
+                  txDetails.buyTx?.value ?? '0',
+                  txDetails.buyAsset?.precision ?? 18
+                )}
+                symbol={txDetails.buyAsset?.symbol ?? ''}
+                maximumFractionDigits={6}
+              />
+            </Row.Value>
+          </Row>
+
+          <Row variant='vertical' hidden={!(txDetails.tx?.fee && txDetails.feeAsset)}>
+            <Row.Label>
+              <Text translation='transactionRow.fee' />
+            </Row.Label>
+            <Row.Value>
+              <Amount.Crypto
+                value={fromBaseUnit(
+                  txDetails.tx?.fee?.value ?? '0',
+                  txDetails.feeAsset?.precision ?? 18
+                )}
+                symbol={txDetails.feeAsset.symbol}
+                maximumFractionDigits={6}
+              />
+            </Row.Value>
+          </Row>
+          <Row variant='vertical'>
+            <Row.Label>
+              <Text translation='transactionRow.status' />
+            </Row.Label>
+            <Row.Value textAlign='left'>
+              {txDetails.tx.status === chainAdapters.TxStatus.Confirmed && (
```

Comment:
Non-blocking, but three conditional renders like this depending on a pattern match is a call for a render function that could use a switch case on `txDetails.tx.status`